src/server/tiered_storage.cc

```diff
@@ -212,11 +185,39 @@ bool TieredStorage::ShardOpManager::ReportFetched(EntryId id, string_view value,
   if (!modified && !cache_fetched_)
     return false;
 
+  // A workaround - to avoid polluting in-memory table by reads that go into a snapshot.
+  // It's not precise because we may handle reads coming from client requests.
+  // TODO: to revisit this when we rewrite it with more efficient snapshotting algorithm.
   if (SliceSnapshot::IsSnaphotInProgress())
     return false;
 
-  SetInMemory(get<OpManager::KeyRef>(id), value, segment);
-  return true;
+  auto key = get<OpManager::KeyRef>(id);
+  auto* pv = Find(key);
```

Comment:
Changed the logic a little bit. Before: SetInMemory returned result but it was unchecked. Now- moved the logic here since it was the only code. Verify it succeeds via `LOG(DFATAL)`